storage/innobase/log/log0log.cc

```diff
@@ -824,64 +842,119 @@ template<bool release_latch> inline lsn_t log_t::write_buf() noexcept
   }
   else
   {
+    ut_ad(write_lock.is_owner());
     ut_ad(!recv_no_log_write);
     write_lock.set_pending(lsn);
     ut_ad(write_lsn >= get_flushed_lsn());
     const size_t block_size_1{get_block_size() - 1};
-    lsn_t offset{calc_lsn_offset(write_lsn) & ~lsn_t{block_size_1}};
+    size_t write_size_1{innodb_log_write_ahead_size - 1};
+    ut_ad(ut_is_2pow(write_size_1 + 1));
+    ut_ad(write_size_1 >= block_size_1);
```

Comment:
We may have `write_size_1<block_size_1` during `srv_start()` when running with 4096-byte sector size. We’d better adjust the `innodb_log_write_ahead_size` during `create_log_file()` to keep this logic simple.

Reply:
Adjusted in 910d3266d9cd0813f3323bd7b4befb4efe19e078 and 61506d9eef08e95458fcba01f5f87c6d41342378, before the calls to `log_sys.set_block_size()`.